### PR TITLE
SNAP-1051 Refactor Snapshot Search UI

### PR DESCRIPTION
**Pull request by ggomez-cambriasolutions**
### Jira Story

https://osi-cwds.atlassian.net/browse/SNAP-1051

## Description
Refactor the Snapshot Search UI per the mockup in the story.

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [ ] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/common/search/age/SearchByAgeMethodSelect.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+SearchByAgeMethodSelect.propTypes = {
+  gridClassName: PropTypes.string,
+  id: PropTypes.string,
+  onChange: PropTypes.func.isRequired,
+  onKeyPress: PropTypes.func,
+  value: PropTypes.string,
+}
+
+export default SearchByAgeMethodSelect
```
Comment (NagaPradeepGajula-osi) [inline app/javascript/common/search/age/SearchByAgeMethodSelect.jsx]: I don't see any tests for this component. May be you missed it in the commits?

Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/age/SearchByAgeMethodSelect.jsx]: Good catch. I'll add it in.

File: app/javascript/common/search/age/AgeSearchFields.jsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      </div>
+    )
+    return shouldRender ? dateOfBirthSection : null
+  }
+
+  renderAgeUnitForm() {
+    const {onChange, onKeyPress, personSearchFields} = this.props
+    return (
+      <AgeUnitForm
+        formLabel = "Unit"
```
Comment (pokharelmanish) [inline app/javascript/common/search/age/AgeSearchFields.jsx]: Let's remove this space for here.

File: app/javascript/common/search/age/AgeSearchFields.jsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        monthsLabel = "Months"
```
Comment (pokharelmanish) [inline app/javascript/common/search/age/AgeSearchFields.jsx]: same

File: app/javascript/common/search/age/AgeSearchFields.jsx [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        yearsLabel = "Years"
```
Comment (pokharelmanish) [inline app/javascript/common/search/age/AgeSearchFields.jsx]: same

**Pull request merged**